pkg/apis/core/v1beta1/validation/validation_test.go

```diff
@@ -358,25 +394,49 @@ func TestValidateKubeFedConfig(t *testing.T) {
 	validFeatureGateConf.Spec.FeatureGates[0].Configuration = v1beta1.ConfigurationMode("NeitherEnableOrDisable")
 	errorCases["spec.featureGates.configuration: Unsupported value"] = validFeatureGateConf
 
+	invalidClusterHealthCheckNil := validKubeFedConfig()
+	invalidClusterHealthCheckNil.Spec.ClusterHealthCheck = nil
+	errorCases["spec.clusterHealthCheck: Required value"] = invalidClusterHealthCheckNil
+
 	zeroInt := int64(0)
 	zeroIntPtr := &zeroInt
 
+	invalidPeriodSecondsNil := validKubeFedConfig()
+	invalidPeriodSecondsNil.Spec.ClusterHealthCheck.PeriodSeconds = nil
+	errorCases["spec.clusterHealthCheck.periodSeconds: Required value"] = invalidPeriodSecondsNil
+
 	validPeriodSecondsGreaterThan0 := validKubeFedConfig()
 	validPeriodSecondsGreaterThan0.Spec.ClusterHealthCheck.PeriodSeconds = zeroIntPtr
 	errorCases["spec.clusterHealthCheck.periodSeconds: Invalid value"] = validPeriodSecondsGreaterThan0
 
+	invalidFailureThresholdNil := validKubeFedConfig()
+	invalidFailureThresholdNil.Spec.ClusterHealthCheck.FailureThreshold = nil
+	errorCases["spec.clusterHealthCheck.failureThreshold: Required value"] = invalidFailureThresholdNil
+
 	validFailureThresholdGreaterThan0 := validKubeFedConfig()
 	validFailureThresholdGreaterThan0.Spec.ClusterHealthCheck.FailureThreshold = zeroIntPtr
 	errorCases["spec.clusterHealthCheck.failureThreshold: Invalid value"] = validFailureThresholdGreaterThan0
 
+	invalidSuccessThresholdNil := validKubeFedConfig()
+	invalidSuccessThresholdNil.Spec.ClusterHealthCheck.SuccessThreshold = nil
+	errorCases["spec.clusterHealthCheck.successThreshold: Required value"] = invalidSuccessThresholdNil
+
 	validSuccessThresholdGreaterThan0 := validKubeFedConfig()
 	validSuccessThresholdGreaterThan0.Spec.ClusterHealthCheck.SuccessThreshold = zeroIntPtr
 	errorCases["spec.clusterHealthCheck.successThreshold: Invalid value"] = validSuccessThresholdGreaterThan0
 
+	validTimeoutSecondsNil := validKubeFedConfig()
+	validTimeoutSecondsNil.Spec.ClusterHealthCheck.TimeoutSeconds = nil
+	errorCases["spec.clusterHealthCheck.timeoutSeconds: Required value"] = validTimeoutSecondsNil
+
 	validTimeoutSecondsGreaterThan0 := validKubeFedConfig()
 	validTimeoutSecondsGreaterThan0.Spec.ClusterHealthCheck.TimeoutSeconds = zeroIntPtr
 	errorCases["spec.clusterHealthCheck.timeoutSeconds: Invalid value"] = validTimeoutSecondsGreaterThan0
 
+	invalidAdoptResourcesNil := validKubeFedConfig()
+	invalidAdoptResourcesNil.Spec.SyncController.AdoptResources = nil
+	errorCases["spec.syncController.adoptResources: Required value"] = invalidAdoptResourcesNil
```

Comment:
Just missing a check for `syncController` being `nil`.

Reply:
Good catch!  Done.